### PR TITLE
Checklist validate import

### PR DESCRIPTION
**Pull request by dczulada**
Validate C1 Record Sample test using Patient import instead of QRDA parsing.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-433
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/validators/checklist_criteria_validator.rb [hunk 1]
```diff
@@ -21,39 +20,109 @@ def initialize(checklist_test)
...
+    end
+
+    # This method will return true if the specified attribute has a stored value
+    def attribute_has_data(attribute, checked_criteria)
+      # If the attribute is a Time, DateTime or an Integer, it was stored correctly
+      return true if attribute.is_a? Time
+      return true if attribute.is_a? DateTime
+      return true if attribute.is_a? Integer
+      # If the attribute is an Array, loop through the array
+      return (attribute.map { |at| attribute_has_data(at, checked_criteria) }.include? true) if attribute.is_a? Array
```
Comment (lmd59) [inline lib/validators/checklist_criteria_validator.rb]: "(attribute.map { |at| attribute_has_data(at, checked_criteria) }.include? true) if attribute.is_a? Array"
suggest: "attribute.any?{ |at| attribute_has_data(at, checked_criteria) } if attribute.is_a? Array" instead

File: lib/validators/checklist_criteria_validator.rb [hunk 2]
```diff
@@ -21,39 +20,109 @@ def initialize(checklist_test)
...
+    def verify_code_attribute(attribute, checked_criteria)
+      # If a attribute_code is specified in the checked_criteria (i.e., filled in the Record Sample Form) make sure it matches
+      return attribute[:code] == checked_criteria['attribute_code'] if checked_criteria['attribute_code']
+
+      # If a attribute_code is not specified in the checked_criteria (i.e., filled in the Record Sample Form) make any code is there
+      !attribute[:code].nil?
+    end
+
+    def verify_component_attribute(attribute, checked_criteria)
+      # A component has nested attributes, check those
```
Comment (lmd59) [inline lib/validators/checklist_criteria_validator.rb]: "A component has nested attributes" -> should we be checking each one?

Comment (lmd59) [reply lib/validators/checklist_criteria_validator.rb]: Re: result only thing truly important
